src/main/java/org/sigmah/offline/dao/UserAsyncDAO.java

```diff
@@ -182,6 +185,66 @@ public void onSuccess(OrgUnitDTO result) {
         });
 	}
 
+public void getByOrgUnit(final int orgUnitId, final AsyncCallback<ListResult<UserDTO>> callback) {
+		openTransaction(Transaction.Mode.READ_ONLY, new OpenTransactionHandler() {
+
+			@Override
+			public void onTransaction(final Transaction transaction) {
+				final ObjectStore userObjectStore = transaction.getObjectStore(Store.USER);
+				final Index orgUnitIndex = userObjectStore.index("orgUnit");
+				final OpenCursorRequest openCursorRequest = orgUnitIndex.openCursor(IDBKeyRange.only(orgUnitId));
+
+				final ArrayList<UserDTO> users = new ArrayList<UserDTO>();
+
+				openCursorRequest.addCallback(new AsyncCallback<Request>() {
+
+					@Override
+					public void onFailure(Throwable caught) {
+						callback.onFailure(caught);
+					}
+
+					@Override
+					public void onSuccess(Request request) {
+						final Cursor cursor = openCursorRequest.getResult();
+						if (cursor != null) {
+							final UserJS userJS = (UserJS) cursor.getValue();
+							users.add(userJS.toDTO());
+							cursor.next();
+
+						} else {
+							callback.onSuccess(new ListResult<UserDTO>(users));
+						}
+					}
+				});
+			}
+		});
+	}
+
+	public void getByOrgUnits(final Set<Integer> orgUnitIds, final AsyncCallback<ListResult<UserDTO>> callback) {
+		final List<UserDTO> users = new ArrayList<UserDTO>();
+		// `i` needs to be final as it is accessed asynchronously
+		final int[] i = {0};
```

Comment:
Maybe we can introduce a simpler `MutableInt` of our own. I think it will be useful in other places of the project as well.